cmd/admin-handlers.go

```diff
@@ -1768,3 +1772,128 @@ func (a adminAPIHandlers) KMSKeyStatusHandler(w http.ResponseWriter, r *http.Req
 	}
 	writeSuccessResponseJSON(w, resp)
 }
+
+// ServiceInfoHandler - GET /minio/admin/v1/serviceinfo?type={serviceType}
+// ----------
+// Get service information
+func (a adminAPIHandlers) ServiceInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ServiceInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	var jsonBytes []byte
+	var err error
+
+	vars := mux.Vars(r)
+
+	switch madmin.InfoType(vars[madmin.INFO]) {
+	case madmin.ETCD:
+		errorString := ""
+		if globalEtcdClient == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrETCDNotConfigured), r.URL)
+			return
+		}
+		etcdEndPoints := globalEtcdClient.Endpoints()
+		var etcdInfos []madmin.ETCDInfo
+		for _, endPoint := range etcdEndPoints {
```

Comment:
We could parallelize these checks using go-routine and `wg.Wait`

Reply:
I dont think there is a need of parallel processing here. In drives and CPU we fetch the info from peers so a parallel processing make sense. Here it is straight forward getting the info using global variables.